spmd/compiler/api.py

```diff
@@ -45,6 +43,14 @@ class Schema:
     placements: List[Placement]
 
 
+def _is_partial_dtensor(obj: object) -> bool:
+    """check if object is 1) DTensor and  2) with placement of _Partial"""
+    is_partial = isinstance(obj, DTensor) and isinstance(
+        obj.placements[0], _Partial
```

Comment:
this will only check if the first placement is partial? I think we should check every placement and return true if there's one partial in the placements?

Reply:
Thanks - changed to walk entire placement list for the _Partial check now.